Tutorial1-1Basics/src/main/java/com/smarttoolfactory/tutorial1_1basics/chapter2_material_widgets/Tutorial2_9_1SideNavigationScaffold.kt

```diff
@@ -42,33 +45,51 @@ import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.tooling.preview.Devices
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 import androidx.navigation.compose.NavHost
 import androidx.navigation.compose.composable
 import androidx.navigation.compose.rememberNavController
 import com.smarttoolfactory.tutorial1_1basics.R
+import com.smarttoolfactory.tutorial1_1basics.isInPreview
 import com.smarttoolfactory.tutorial1_1basics.ui.ComposeTutorialsTheme
 import com.smarttoolfactory.tutorial1_1basics.ui.components.DrawerButton
 import kotlinx.coroutines.launch
 
 @Preview
 @Composable
-fun Tutorial2_9Screen1() {
-    TutorialContent()
+fun Tutorial2_9Screen1(
+    @PreviewParameter(DrawerStateProvider::class)
+    drawerValue: DrawerValue
+) {
+    TutorialContent(
+        scaffoldState = rememberScaffoldState(
+            drawerState = rememberDrawerState(initialValue = drawerValue)
```

Comment:
Providing different initial values for DrawerState. This can also be utilized to provide custom `SnackbarHostState`.


<img width="538" alt="Screenshot 2024-03-05 at 7 59 25 AM" src="https://github.com/SmartToolFactory/Jetpack-Compose-Tutorials/assets/43310446/e1667697-c249-41e1-aa79-c83dcb39ed5d">